src/App.tsx

```diff
@@ -1,20 +1,25 @@
 import React from 'react';
 import Header from './components/Header';
 import PokemonList from './components/PokemonList';
-import { PokemonDataProvider } from './contexts/PokemonDataProvider';
 import './styles/global.css';
 import './App.css';
 import { FilterProvider } from './contexts/FilterProvider';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { ReactQueryDevtools } from '@tanstack/react-query-devtools';
 
 const App: React.FC = () => {
+  const queryClient = new QueryClient();
```

Comment:
Quando vc cria o `QueryClient`, pode passar algumas configurações default pra ele. Uma que é bastante comum é parar de refetch sempre que abrir a aba novamente:
```suggestion
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        refetchOnWindowFocus: false,
      },
    },
  });
```
Uma nota importante é que as opções passadas pra cada `query` tem prioridade sobre essas de configuração inicial, por exemplo:
```ts
useQuery('someQueryKey', fetchSomeData, {
  refetchOnWindowFocus: true,
});
```

No caso, a opção da `query` como `true` iria prevalecer. Isso é importante de saber principalmente na hora de testar, quando a gente usa um `QueryClient` mockado, passa alguma config, espera um resultado e aí não acontece o esperado (muito provavelmente é porque a `useQuery` tá sobrescrevendo o `queryClient`).